addon/snapshot.js

```diff
@@ -57,6 +57,28 @@ function setTextareaContent(dom) {
   return dom;
 }
 
+function serializeCanvasElements(originalDOM, clonedDOM) {
+  let originals = originalDOM.find('canvas');
+  let clones = clonedDOM.find('canvas');
+
+  for (let i = 0; i < originals.length && clones.length; i++) {
+    let original = originals[i];
+    let clone = clones[i];
+
```

Comment:
Hey, I had to put:

```javascript
if (original == null || clone == null) { break; }
```

in this spot to make this work in my testing. It looks like `clone` can sometimes be `null` and that was causing problems down the line.

Can you update your PR to have this line?

The way I tested this was just add this test code to an ember app:

```html
  <div>
    <canvas id="myCanvas" width="200" height="100">
    </canvas>
    <canvas id="myCanvas2" width="200" height="100">
    </canvas>
  </div>
  <script>
    var c = document.getElementById("myCanvas");
    var ctx = c.getContext("2d");

    // Create gradient
    var grd = ctx.createRadialGradient(75, 50, 5, 90, 60, 100);
    grd.addColorStop(0, "red");
    grd.addColorStop(1, "white");

    // Fill with gradient
    ctx.fillStyle = grd;
    ctx.fillRect(10, 10, 150, 80);
  </script>
```

This is the error I was getting, until I added the line I'm proposing we add:

```
not ok 1 Chrome 68.0 - Acceptance: Marketing pages user is unauthenticated can visit /
    ---
        message: >
            Cannot read property 'parentNode' of undefined
        stack: >
            TypeError: Cannot read property 'parentNode' of undefined
                at serializeCanvasElements (http://localhost:7357/assets/vendor.js:334492:13)
                at percySnapshot (http://localhost:7357/assets/vendor.js:334551:22)
                at Class.<anonymous> (http://localhost:7357/assets/tests.js:3311:35)
                at http://localhost:7357/assets/vendor.js:65374:19
                at isolate (http://localhost:7357/assets/vendor.js:66552:17)
                at onFulfillment (http://localhost:7357/assets/vendor.js:66489:16)
                at tryCatcher (http://localhost:7357/assets/vendor.js:74122:21)
                at invokeCallback (http://localhost:7357/assets/vendor.js:74294:33)
                at http://localhost:7357/assets/vendor.js:74358:16
                at http://localhost:7357/assets/vendor.js:65388:9
        Log: |
```

Reply:
@djones I tried to reproduce that issue but I couldn't and I'm not sure yet how that could happen in general. I've pushed my reproduction attempt to https://github.com/Turbo87/percy-canvas-test, maybe you can spot a difference to what you were doing?

putting in that guard will certainly be better than crashing, but I'd prefer to find the root cause of the issue instead 🤔

Reply:
@djones any thoughts?